python/cudf_polars/cudf_polars/dsl/expr.py

```diff
@@ -0,0 +1,1023 @@
+        self,
+        df: DataFrame,
+        *,
+        context: ExecutionContext = ExecutionContext.FRAME,
+        mapping: dict[Expr, Column] | None = None,
+    ) -> Column:
+        """Evaluate this expression given a dataframe for context."""
+        values, mask = (
+            child.evaluate(df, context=context, mapping=mapping)
+            for child in self.children
+        )
+        table = plc.stream_compaction.apply_boolean_mask(
+            plc.Table([values.obj]), mask.obj
+        )
+        return Column(table.columns()[0], values.name).with_sorted(like=values)
+
+    def collect_agg(self, *, depth: int) -> AggInfo:
+        """Collect information about aggregations in groupbys."""
+        # TODO: Could do with sort-based groupby and segmented filter
+        raise NotImplementedError("Filter in groupby")
+
+
+class RollingWindow(Expr):
+    __slots__ = ("options", "children")
+    _non_child = ("dtype", "options")
+
+    def __init__(self, dtype: plc.DataType, options: Any, agg: Expr):
+        super().__init__(dtype)
+        self.options = options
+        self.children = (agg,)
+
+
+class GroupedRollingWindow(Expr):
+    __slots__ = ("options", "children")
+    _non_child = ("dtype", "options")
+
+    def __init__(self, dtype: plc.DataType, options: Any, agg: Expr, *by: Expr):
+        super().__init__(dtype)
+        self.options = options
+        self.children = (agg, *by)
+
+
+class Cast(Expr):
+    __slots__ = ("children",)
+    _non_child = ("dtype",)
+
+    def __init__(self, dtype: plc.DataType, value: Expr):
+        super().__init__(dtype)
+        self.children = (value,)
+
+    @with_mapping
+    def evaluate(
+        self,
+        df: DataFrame,
+        *,
+        context: ExecutionContext = ExecutionContext.FRAME,
+        mapping: dict[Expr, Column] | None = None,
+    ) -> Column:
+        """Evaluate this expression given a dataframe for context."""
+        (child,) = self.children
+        column = child.evaluate(df, context=context, mapping=mapping)
+        return Column(plc.unary.cast(column.obj, self.dtype), column.name).with_sorted(
+            like=column
+        )
+
+    def collect_agg(self, *, depth: int) -> AggInfo:
+        """Collect information about aggregations in groupbys."""
+        # TODO: Could do with sort-based groupby and segmented filter
+        (child,) = self.children
+        return child.collect_agg(depth=depth)
+
+
+class Agg(Expr):
+    __slots__ = ("name", "options", "op", "request", "children")
+    _non_child = ("dtype", "name", "options")
+
+    def __init__(
+        self, dtype: plc.DataType, name: str, options: Any, value: Expr
+    ) -> None:
+        super().__init__(dtype)
+        self.name = name
+        self.options = options
+        self.children = (value,)
+        if name not in Agg._SUPPORTED:
+            raise NotImplementedError(f"Unsupported aggregation {name=}")
+        # TODO: nan handling in groupby case
+        if name == "min":
```

Comment:
Maybe could be simplified a bit using a getattr + a dict lookup for cases where the names don't match exactly (like var->variance)? Something like
```
try:
    req = getattr(plc.aggregation, {"var": "variance"}.get(name, name))(agg_arg_map[name])
except KeyError, AttributeError:
    ....
```
Just a suggestion, don't have to use it.